src/OnlineSales/Migrations/20230126073650_AccountEntity.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Metadata;
+
+#nullable disable
+
+namespace OnlineSales.Migrations
+{
+    /// <inheritdoc />
+    public partial class AccountEntity : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "account",
+                columns: table => new
+                {
+                    id = table.Column<int>(type: "integer", nullable: false)
+                        .Annotation("Npgsql:ValueGenerationStrategy", NpgsqlValueGenerationStrategy.IdentityByDefaultColumn),
+                    name = table.Column<string>(type: "text", nullable: false),
+                    city = table.Column<string>(type: "text", nullable: true),
+                    statecode = table.Column<string>(name: "state_code", type: "text", nullable: true),
+                    country = table.Column<string>(type: "text", nullable: true),
+                    employeesrate = table.Column<string>(name: "employees_rate", type: "text", nullable: true),
+                    revenue = table.Column<double>(type: "double precision", nullable: true),
+                    domainid = table.Column<int>(name: "domain_id", type: "integer", nullable: false),
+                    tags = table.Column<string[]>(type: "jsonb", nullable: false),
+                    socialmedia = table.Column<Dictionary<string, string>>(name: "social_media", type: "jsonb", nullable: true),
+                    data = table.Column<string>(type: "jsonb", nullable: true),
+                    createdat = table.Column<DateTime>(name: "created_at", type: "timestamp with time zone", nullable: false),
+                    createdbyip = table.Column<string>(name: "created_by_ip", type: "text", nullable: true),
+                    createdbyuseragent = table.Column<string>(name: "created_by_user_agent", type: "text", nullable: true),
+                    updatedat = table.Column<DateTime>(name: "updated_at", type: "timestamp with time zone", nullable: true),
+                    updatedbyip = table.Column<string>(name: "updated_by_ip", type: "text", nullable: true),
+                    updatedbyuseragent = table.Column<string>(name: "updated_by_user_agent", type: "text", nullable: true)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("pk_account", x => x.id);
+                    table.ForeignKey(
+                        name: "fk_account_domain_domain_id",
+                        column: x => x.domainid,
+                        principalTable: "domain",
+                        principalColumn: "id",
+                        onDelete: ReferentialAction.Cascade);
```

Comment:
Please make it restricted here as well. Domain should not be deleted when you delete a contact and visa versa.